### PR TITLE
Allow Spore environment variables to point to other environment variables

### PR DESCRIPTION
**Pull request by marcinwyszynski**

### CODE REVIEW

File: lib/spore/environment.rb [hunk 1]
```diff
@@ -17,11 +17,20 @@ def read
...
 
     def apply!
-      each { |k, v| ENV[k] = v }
+      each { |k, v| ENV[k] = maybe_indirect(v) }
+    end
+
+    private
+
+    # Indirect values point to other environment variables and start with two
+    # tildes (~~).
```
Comment (sguha00) [inline lib/spore/environment.rb]: Is the double tilde `~~` a standard way to refer to other environment variables?

Comment (marcinwyszynski) [inline lib/spore/environment.rb]: I kind of made it up but yes, that's the intention. My thinking behind it is that allowing full ability to eval an arbitrary piece of Ruby or Shell is a potential security vulnerability and using $ as an indicator of another instance variable could yield false positives. So double tilde it is. Feel free to suggest an alternative, I don't really have a strong opinion on what it should be as long as I have this option ;)

**Comment by marcinwyszynski** [comment]
I actually forgot to anchor double tilde to the beginning of the value, fixed now.

File: lib/spore/environment.rb (hunk 1 above)
Comment (treygriffith) [inline lib/spore/environment.rb]: I think I'd prefer to keep it more consistent with `dotenv`. In that library, they do parsing using this substitutions module https://github.com/bkeepers/dotenv/blob/master/lib/dotenv/substitutions/variable.rb

Comment (marcinwyszynski) [inline lib/spore/environment.rb]: No bother, I'll take the same approach and while at it fix the bug in dotenv's code where unbalanced curly braces around the environment variable would still trigger a substitution.

**Comment by marcinwyszynski** [comment]
OK folks, PTAL. I added full specs for this new functionality, too.

**Comment by treygriffith** [comment]
This should probably be a minor version instead of a patch version bump, but other than that it looks good to me. Your thoughts @sguha00 ?

**Comment by sguha00** [comment]
Looks good, but I agree with @treygriffith, seems like it's a big enough change to bump the minor version

**Comment by treygriffith** [comment]
@sguha00 would you mind merging and publishing on ruby gems?

**Comment by marcinwyszynski** [comment]
There, fixed.

**Comment by marcinwyszynski** [comment]
Ping?

**Pull request merged**